### PR TITLE
Write DBNull to data type of Array(T) 

### PR DESCRIPTION
**Pull request by powerking**

### CODE REVIEW

**Comment by smbecker** [comment]
I am running into this same issue when trying to import data into CH from Npgsql. This PR has been sitting for quite a while. What is needed to get it merged in and how can I help?

File: ClickHouse.Client/Types/ArrayType.cs [hunk 1]
```diff
@@ -38,6 +38,11 @@ public override object Read(ExtendedBinaryReader reader)
 
         public override void Write(ExtendedBinaryWriter writer, object value)
         {
+            if (value == DBNull.Value || value == null)
+            {
+                UnderlyingType.Write(writer, 0);
```
Comment (smbecker) [inline ClickHouse.Client/Types/ArrayType.cs:43]: This should be `writer.Write7BitEncodedInt(0);`

**Comment by DarkWanderer** [comment]
> I am running into this same issue when trying to import data into CH from Npgsql. This PR has been sitting for quite a while. What is needed to get it merged in and how can I help?

You can just recreate it, this particular PR looks to be abandoned

**Comment by DarkWanderer** [comment]
Implemented in #326

**Pull request closed without merging**